tensorboard/program.py

```diff
@@ -520,21 +521,25 @@ def handle_error(self, request, client_address):
       logger.error('HTTP serving error', exc_info=exc_info)
 
   def get_url(self):
-    if self._auto_wildcard:
-      display_host = socket.gethostname()
-    else:
-      host = self._flags.host
-      display_host = (
+    if not self.display_host:
```

Comment:
Let's make the variable `self._url` instead and just store the full URL.

It would probably read better if we factor out something like `get_url_host()` e.g.

```python
def get_url_host(self):
  if not self._auto_wildcard:
    return self._flags.host
  host = socket.getfqdn()
  try:
    socket.create_connection((host, self.server_port), timeout=1)
    return host
  except socket.error as e:
    return 'localhost'

def get_url(self):
  if not self._url:
    host = self.get_url_host()
    display_host = (
        '[%s]' % host if ':' in host and not host.startswith('[') else host)
    self._url = 'http://%s:%d%s/' % (self.display_host, self.server_port,
        self._flags.path_prefix.rstrip('/'))
  return self._url
```